File: leetcode/easy/java/lc121.java
class Solution {
    public int maxProfit(int[] prices) {
        int maximum = 0;
        int minPrice = prices[0];
        for (int i=0; i<prices.length; i++) {
            maximum = Math.max(maximum, prices[i] - minPrice);
            minPrice = Math.min(minPrice, prices[i]);
        }
        return maximum;
    }
}

File: leetcode/easy/go/lc121.go
func maxProfit(prices []int) int {
    if len(prices) == 0 {
        return 0
    }
    currMin := prices[0]
    maxProfit := 0
    for i := 1; i < len(prices); i++ {
        if prices[i] < currMin {
            currMin = prices[i]
        } else {
            profit := prices[i] - currMin
            if profit > maxProfit {
                maxProfit = profit
            }
        }
    }
    return maxProfit
}

